lxd/instance_snapshot.go

```diff
@@ -523,11 +523,11 @@ func snapshotPut(s *state.State, r *http.Request, snapInst instance.Instance) re
 	}
 
 	opType := operationtype.SnapshotUpdate
+	snapName := snapInst.Name()
+	parentName, _, _ := api.GetParentAndSnapshotName(snapName)
 
-	parentName, _, _ := api.GetParentAndSnapshotName(snapInst.Name())
-
-	resources := map[string][]string{}
-	resources["instances"] = []string{parentName}
+	resources := map[string][]api.URL{}
+	resources["instances"] = []api.URL{*api.NewURL().Path(version.APIVersion, "instances", parentName, "snapshots", snapName)}
```

Comment:
Won't this result in a URL like:

```
/1.0/instances/c1/snapshots/c1%2snap0
```

When it should be:

```
/1.0/instances/c1/snapshots/snap0
```

I think you need to take the 2nd return value from `GetParentAndSnapshotName` and use it here instead of `snapInst.Name()`.